synapse/rest/client/v1/room.py

```diff
@@ -745,7 +745,7 @@ async def on_GET(self, request):
             if server:
                 raise e
 
-        limit = parse_integer(request, "limit", 0)
+        limit: Optional[int] = parse_integer(request, "limit", 0)
```

Comment:
We need to specify now that this can be `Optional` since the `parse_integer` call always returns an `int`, which mypy now realizes. We manually set it to `None` below if `limit == 0`.